app/src/components/market/steps/items/create_market_step.tsx

```diff
@@ -204,9 +197,11 @@ const CreateMarketStep = (props: Props) => {
         <Button
           buttonType={ButtonType.primary}
           disabled={
-            (!MarketCreationStatus.is.ready(marketCreationStatus) &&
-              !MarketCreationStatus.is.error(marketCreationStatus)) ||
-            (!hasEnoughBalanceToFund && !!account)
+            !(
+              MarketCreationStatus.is.ready(marketCreationStatus) &&
+              !MarketCreationStatus.is.error(marketCreationStatus) &&
+              hasEnoughBalance
+            ) && !!account
```

Comment:
This is too hard to read. I think the problem is that we are merging two buttons into one. A better approach might be something like:

```typescript
account ?
<Button disabled={!MarketCreationStatus.is.ready(marketCreationStatus) || ...} ... >Create</Button>
: <Button ...>Connect Wallet</Button>
```

They are going to have some props in common, but that's probably fine.

An alternative is to extract this in a variable:

```
const isButtonDisabled = account ? (!MarketCreation.is.ready(...) ...) : false
```

I think I prefer the first approach, but I'm ok with either one.

Reply:
I opted for split them in two